_rules/__tests__/test-assets.js

```diff
@@ -0,0 +1,101 @@
+const path = require('path')
+const assert = require('assert')
+const globby = require('globby')
+const gfmCodeBlocks = require('gfm-code-blocks')
+const isUrl = require('is-url')
+
+const describeRule = require('../../test-utils/describe-rule')
+const getAllMatchesForRegex = require('../../utils/get-all-matches-for-regex')
+const allTestAssetPaths = uniqueArrayItems(globby.sync(`./test-assets/**/*.*`).map(path => path.replace(`./`, ``)))
+
+describeRule('check if referenced test assets exists', ruleData => {
+	const codeBlocks = gfmCodeBlocks(ruleData.body)
+	const usedAssetPaths = getUniqueTestAssetsUsedInCodeBlocks(codeBlocks)
+
+	if (usedAssetPaths && usedAssetPaths.length) {
+		test.each(usedAssetPaths)('%s', assetPath => {
+			const message = `Asset at path -> ${assetPath} does not exist`
+			expect(allTestAssetPaths.includes(assetPath), message).toBe(true)
+		})
+	}
+})
```

Comment:
I think we should also do the reverse test (check that files in the `test-assets` directory are used somewhere) in order to clean up old test assets that are not used anymore (if any).

Given that you already have `allTestAssetPaths` and `usedAssedPaths`, that should be fairly easy.